src/main/host/host.rs

```diff
@@ -1,7 +1,33 @@
+use once_cell::unsync::OnceCell;
+use std::net::IpAddr;
+use std::sync::Arc;
+
 use crate::cshadow;
 
+#[derive(Debug, PartialEq, Eq, Hash, Copy, Clone)]
+pub struct HostId(u32);
+
+/// Immutable information about the Host.
+#[derive(Debug, Clone)]
+pub struct HostInfo {
+    pub id: HostId,
+    pub name: String,
+    pub default_ip: IpAddr,
+    pub log_level: Option<log::LevelFilter>,
+}
+
 pub struct Host {
     chost: *mut cshadow::Host,
+
+    // Store immutable info in an Arc, that we can safely clone into the Worker
+    // and ShadowLogger.
+    //
+    // Created on-demand for now, to avoid building unnecessarily for ephemeral
+    // Host objects.
+    //
+    // TODO: Consider caching a couple copies that we can "lend" out by value,
+    // without having to do any atomic operation?
+    info: OnceCell<Arc<HostInfo>>,
 }
```

Comment:
A comment here might be useful to explain that the Rust `Host` is meant to be ephemeral, and there isn't a 1:1 relationship with a corresponding C `Host` (for the lifetime of a C `Host`, there will be many Rust `Host`s that point to it).